### DISCUSSION TITLE
WARNING: ignoring settings from 'local.properties'!

### DISCUSSION THREAD

**Discussion by benji-farquhar** [codemagic-ci-cd/codemagic-docs#1928] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Build Android in the workflow editor with release mode and args: --flavor qa -t lib/main_qa.dart --release --no-tree-shake-icons

### Expected results

It succeeds

### Actual results

> WARNING: ignoring settings from 'local.properties'!

displays at the end of the get dependencies task. Then

> Cannot invoke method toInteger() on null object

displays at the end of the build android task.

Cannot invoke method toInteger() on null object happens when your minSdkVersion is below 21 and I set mine to 21 in android/local.properties so it needs to read that file and apply it to the build but it isn't.

I added this to the post clone script, taken from the flutter android and ios sample project:

```
#!/bin/sh
echo "flutter.sdk=$HOME/programs/flutter" > "$CM_BUILD_DIR/android/local.properties"
```

Which should apply the local.properties.

### Build id (optional)

64c53ea14abc96b5dce1091c

**Comment by kulakarla** [comment]
@BenjiFarquhar 

Hello Benji, minSdkVersion should be changed in `/android/app/build.gradle` under `defaultConfig` section.
Change it and let me know if you are still running into the same issue.

**Comment by mikhail-tokarev** [comment]
it was answered on StackOverflow https://stackoverflow.com/questions/76791779/codemagic-warning-ignoring-settings-from-local-properties
